dashboard/src/components/Filter/CodeBlock.tsx

```diff
@@ -1,20 +1,147 @@
+
+import { FormattedMessage } from 'react-intl';
+
+import { useMemo } from 'react';
+
 import { cn } from '@/lib/utils';
 
+import { Tooltip, TooltipTrigger, TooltipContent } from '@/components/Tooltip';
+
+import ColoredCircle from '@/components/ColoredCircle/ColoredCircle';
+import { formattedBreakLineValue } from '@/locales/messages';
+
 type TCodeBlockProps = {
   code: string;
   className?: string;
+  highlightsClassname?: string;
 };
 
-const CodeBlock = ({ code, className }: TCodeBlockProps): JSX.Element => {
+interface IHighlightedCode {
+  highlightedCode: string;
+  highlightCount: number;
+  failCount: number;
+  errorCount: number;
+}
+
+const CodeBlock = ({
+  code,
+  className,
+  highlightsClassname,
+}: TCodeBlockProps): JSX.Element => {
+  const highlightedCode: IHighlightedCode = useMemo(() => {
+    let highlights = 0;
+    let fails = 0;
+    let errors = 0;
+
+    // matches any < or > and replaces with equivalent &lt or &gt
+    let newCode = code.replace(/[<>]/gm, match => {
+      match = match.replace(/</, '&lt');
+      match = match.replace(/>/, '&gt');
+      return match;
+    });
+
+    newCode = newCode.replace(
+      // matches any line with the occurrence of error or fail
+      /^.*(error|fail).*$/gim,
+      match => {
+        highlights++;
+        if (
+          // matches failed to/with, more than 0 fails/faileds and no flags
+          match.search(
+            /.*((\bfailed(\s*to|\s*with|([\b\s:]\s*\(*-*[1-9])))|fail(\b|:|\s*[1-9])(?![|/]))/i,
+          ) !== -1
+        ) {
+          fails++;
+          return '<span class="text-red">' + match + '</span>';
+        }
+        // matches error codes greater than 0 or more than 0 errors
+        if (
+          match.search(
+            /(error(:|,|[\b\s:]\s*\(*-*[1-9a-z]))|([1-9]\s*error)/i,
+          ) !== -1
+        ) {
+          errors++;
+          return '<span class="text-orange-500">' + match + '</span>';
+        }
+        return '<span class="text-sky-600">' + match + '</span>';
+      },
+    );
+    return {
+      highlightedCode: newCode,
+      highlightCount: highlights,
+      failCount: fails,
+      errorCount: errors,
+    };
+  }, [code]);
+
   return (
-    <pre
-      className={cn(
-        'max-h-[275px] w-full max-w-[calc(100vw_-_398px)] overflow-x-auto bg-[#DDDDDD] p-4 font-mono text-[#767676]',
-        className,
+    <div>
+      <pre
+        className={cn(
+          'max-h-[275px] w-full max-w-[calc(100vw_-_398px)] overflow-x-auto bg-[#DDDDDD] p-4 font-mono text-[#767676]',
+          className,
+        )}
+      >
+        <div
+          dangerouslySetInnerHTML={{ __html: highlightedCode.highlightedCode }}
+        ></div>
+      </pre>
+      {highlightedCode.highlightCount > 0 && (
+        <ul className={cn('flex gap-2 py-4 pl-3', highlightsClassname)}>
+          <li>
+            <span className="flex items-center gap-1">
+              <Tooltip>
+                <TooltipTrigger>
```

Comment:
Ana's comment can probably be solved by passing `asChild` as a prop here, it is worth to give it a try

Reply:
If I use asChild it loses the tooltip interactivity, it is also not being used in any of the tooltips in the dashboard